src/kOS/Suffixed/VesselTarget.Parts.cs

```diff
@@ -0,0 +1,392 @@
+                if (allParts == null)
+                    ConstructParts();
+                return rootPart;
+            }
+        }
+        public DecouplerValue NextDecoupler
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return nextDecoupler;
+            }
+        }
+        public ListValue<PartValue> Parts
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return allParts;
+            }
+        }
+        public ListValue<DockingPortValue> DockingPorts
+        {
+            get
+            {
+                if (dockingPorts == null)
+                    ConstructParts();
+                return dockingPorts;
+            }
+        }
+        public ListValue<DecouplerValue> Decouplers
+        {
+            get
+            {
+                if (decouplers == null)
+                    ConstructParts();
+                return decouplers;
+            }
+        }
+        public PartValue this[global::Part part]
+        {
+            get
+            {
+                if (allParts == null)
+                    ConstructParts();
+                return partCache[part];
+            }
+        }
+        private void ConstructParts()
+        {
+            rootPart = null;
+            nextDecoupler = null;
+            allParts = new ListValue<PartValue>();
+            dockingPorts = new ListValue<DockingPortValue>();
+            decouplers = new ListValue<DecouplerValue>();
+            partCache = new Dictionary<global::Part, PartValue>();
+
+            ConstructPart(Vessel.rootPart, null, null);
+
+            allParts.IsReadOnly = true;
+            dockingPorts.IsReadOnly = true;
+            decouplers.IsReadOnly = true;
+        }
+        private void ConstructPart(global::Part part, PartValue parent, DecouplerValue decoupler)
+        {
+            if (part.State == PartStates.DEAD || part.transform == null)
+                return;
+
+            PartValue self = null;
+            foreach (var module in part.Modules)
+            {
+                if (module is IEngineStatus engine)
+                {
+                    self = new EngineValue(Shared, part, parent, decoupler);
+                    break;
+                }
+                if (module is IStageSeparator)
+                {
+                    if (module is ModuleDockingNode dock)
+                    {
+                        var port = new DockingPortValue(Shared, part, parent, decoupler, dock);
+                        self = port;
+                        dockingPorts.Add(port);
+                        if (dock.stagingEnabled)
+                        {
+                            decoupler = port;
+                            decouplers.Add(decoupler);
+                        }
+                    }
+                    else
+                    {
```

Comment:
I would like to add
```
                        if (!module.StagingEnabled())
                            break;
```
here, because:
1. It now matches heat shields, which is causing unexpected results - `stage:nextDecoupler` can return that heat shield instead of `None`.
2. `:DECOUPLER`/`:NEXTDECOUPLER` should not match any decoupler with staging disabled, not even launch clamps.

The question is:
Should everything with staging disabled be ignored? (that is the proposed change)
Or should the part still get the type but not be couned as decoupler (seen in `:DECOUPLER`/`:NEXTDECOUPLER`)?
The second option would be quite problematic. Is heat shield a decoupler or not?
Is decoupler with disabled staging still a decoupler?
Do you want to see launch clamp with disabled staging as launch clamp?